extensions-core/google-extensions/src/main/java/org/apache/druid/storage/google/GoogleStorage.java

```diff
@@ -66,6 +67,11 @@ public InputStream get(final String bucket, final String path, long start) throw
     return inputStream;
   }
 
+  public StorageObject getMetadata(final String bucket, final String path) throws IOException
+  {
+    return storage.get().objects().get(bucket, path).execute();
```

Comment:
> Other paths retry. Should this one?

Hmm. It seems like some other paths retry and some don't. The Google SDK has built-in and configurable retries: https://cloud.google.com/storage/docs/retry-strategy#java. We don't customize this when creating SDK clients, and the defaults do provide for what looks like a reasonable level of retries (6 attempts with backoff over about a minute). So that suggests we don't need to do our own.

> What happens if the object does not exist? Is the intent to fail the ingest? Or, skip missing files? If skip, how will the user know which files were skipped?

If the object doesn't exist, the intent is that the ingest fails.

Reply:
Thanks for the explanation.